src/shell/js/composeui-fdc3/src/ComposeUIChannel.ts

```diff
@@ -0,0 +1,69 @@
+/* 
+ *  Morgan Stanley makes this available to you under the Apache License,
+ *  Version 2.0 (the "License"). You may obtain a copy of the License at
+ *       http://www.apache.org/licenses/LICENSE-2.0.
+ *  See the NOTICE file distributed with this work for additional information
+ *  regarding copyright ownership. Unless required by applicable law or agreed
+ *  to in writing, software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ *  or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ *  
+ */
+
+import { Channel, Context, ContextHandler, DisplayMetadata, Listener } from "@finos/fdc3";
+import { MessageRouter } from "@morgan-stanley/composeui-messaging-client";
+import { ChannelType } from "./ChannelType";
+import { ComposeUIListener } from "./ComposeUIListener";
+import { Fdc3ChannelMessage } from "./Fdc3ChannelMessage";
+import { ComposeUITopic } from "./ComposeUITopic";
+
+export class ComposeUIChannel implements Channel{
+    id!: string;
+    type!: "user" | "app" | "private";
+    displayMetadata?: DisplayMetadata | undefined;
```

Comment:
@kruplm what's your opinion on this? I think an optional property or parameter of type `T?` implicitly has the type `T | undefined`, so you don't need `| undefined`.

Reply:
I can remove the undefined, the IDE did add that automatically :D

Reply:
That's why I added @kruplm , I'm not 100% sure my assumption is correct